lua/nvim-tree/explorer/sorters.lua

```diff
@@ -298,13 +238,79 @@ function C.filetype(a, b)
   return a_ft < b_ft
 end
 
-function M.setup(opts)
-  M.config = {}
-  M.config.sort = opts.sort
+---@class Sorter
+local Sorter = {}
+
+function Sorter:new(opts)
+  local o = {} -- create object if user does not provide one
+  setmetatable(o, self)
+  self.__index = self
+  o.config = opts.sort
```

Comment:
We're going to need to take a deep copy here so that we don't mutate the global user setup options.

`o.config = vim.deepcopy(opts.sort)`